### PR TITLE
Polkassembly link to proposal components

### LINKED ISSUE #1032: Add PolkAssembly link to proposal (democracy) and referendum  + make it cohesive along the app
> As a user I wanna easily find the origin of a proposal and the discussions around it

_[more details.]_

We already have this feature for council motions and treasury proposals but they are presented differently. Let's make it uniform.

For the democracy proposal the referendum, the treasury proposal, and council motions **teasers** let's reuse the UI already implemented on treasury proposals:

For the council motion, democracy proposal and referendum **details screens** let's use the UI already implemented on motion details screen:

Summary of what has to be implemented:

- [ ] Add Polkassembly redirection link on Motions, democracy proposal and referendum teasers
image

- [ ] Add "details on PolkAssembly" on democracy proposal and referendum details screens
image

### PR DESCRIPTION
**Pull request by ishrak-imam**
### Description

Fixes: #1032 

Added polkassembly link to proposal item components

<Image src="https://user-images.githubusercontent.com/16683923/161722833-c9ebab5c-6eca-4da7-a2c6-53053edb5f2f.png" width=250 /> <Image src="https://user-images.githubusercontent.com/16683923/161722838-a33d432b-1513-4043-bb86-f27484c8419f.png" width=250 /> <Image src="https://user-images.githubusercontent.com/16683923/161722809-0e628430-30dd-42c9-899a-02a8f56d4537.png" width=250 /> <Image src="https://user-images.githubusercontent.com/16683923/161726472-ce067ddd-fdb7-4e4a-a694-80f26ef3bf60.png" width=250 /> <Image src="https://user-images.githubusercontent.com/16683923/161726479-a481dc6c-5d66-478b-96fd-cbe53185168b.png" width=250 />

### CODE REVIEW

**Comment by silva-fj** [comment]
Is there any noticeable difference in the UI? Can you add some screenshots ?

File: src/ui/screens/DemocracyScreen.tsx [hunk 1]
```diff
@@ -99,6 +122,48 @@ export function DemocracyScreen() {
...
 }
 
+type ProposalTeaserProps = {
+  proposal: Proposal;
+  children?: React.ReactNode;
+  navigation: NavigationProp<DashboardStackParamList>;
+};
+
+function DemocracyProposalTeaser({proposal, children, navigation}: ProposalTeaserProps) {
+  if (proposal.__typename === 'SubstrateChainDemocracyReferendum') {
```
Comment (silva-fj) [inline src/ui/screens/DemocracyScreen.tsx:132]: NIT: How about if we create a utility function like `function getTypeFrom(proposal: Proposal) =>  'proposal' | 'referendum';` ?

File: src/utils/proposal.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type {DemocracyProposal, DemocracyReferendum, ProposalArg} from 'src/api/hooks/useDemocracy';
+import type {MotionProposal} from 'src/api/hooks/useCouncilMotions';
+import {stringShorten} from '@polkadot/util';
+
+type Proposal = DemocracyProposal | DemocracyReferendum | MotionProposal;
+
+export function getProposalTitle(proposal: Proposal) {
+  if (proposal.method && proposal.section) {
+    return `${proposal.method}.${proposal.section}()`;
+  } else if (proposal.__typename === 'SubstrateChainDemocracyReferendum') {
```
Comment (silva-fj) [inline src/utils/proposal.ts:10]: The utils can also be reused here to make it a bit more readable

File: src/ui/screens/DemocracyScreen.tsx (hunk 1 above)
Comment (ishrak-imam) [reply src/ui/screens/DemocracyScreen.tsx:132]: In that case we loose the TS union discrimination. Is there a way to make TS still infer the types using a util function?

Comment (silva-fj) [reply src/ui/screens/DemocracyScreen.tsx:132]: True, that was the point of using getting a `union` type from the api so we don't have to create unions in the client and use `__typename`. But for now we can live with that

**Review by silva-fj — APPROVED** [review]

**Pull request merged**